### ISSUE TITLE
pnpm-lock.yaml not removed after pnpmEnable is set back to false

### ISSUE THREAD

**Issue by web-padawan** [vaadin/flow#7461]
## Description

Switching to `pnpm` removes `package-lock.json` (works as expected).
But switching back to `npm` does NOT remove `pnpm-lock.yaml`.

## Steps to reproduce

1. download project from https://start.vaadin.com/ with empty view

2. execute `mvn spring-boot:run`

3. update `pom.xml` to enable npm:
```xml
  <configuration>
      <pnpmEnable>true</pnpmEnable>
  </configuration>
```

4. run `rm -rf node_modules target && mvn spring-boot:run -Pproduction`:
  - `pnpm-lock.yaml` was created
  - `package-lock.json` was removed

5. remove `<pnpmEnable>true</pnpmEnable>` setting from `pom.xml`.

6. run `rm -rf node_modules target && mvn spring-boot:run -Pproduction`:
  - `package-lock.json` is restored
  - `pnpm-lock.yaml` is NOT removed

**Comment by pleku** [comment]
We should remove the `pnpm-lock.yaml` when we notice it is there but npm is used due to `pnpm.enable=false`, since leaving it there could just confuse the users.

**Comment by denis-anisimov** [comment]
That's quite safe: if npm is used (we know when it's used) just remove `pnpm-lock.yaml` every time.

**Comment by web-padawan** [comment]
This was fixed by #7543 but not automatically closed due to a typo. Closing as done.
